### PR TITLE
feat(chatbar): undo/redo manager

### PR DESCRIPTION
**Pull request by molimauro**
**What this PR does** 📖
Add undo/redo functionalities to the chatbar.
**Which issue(s) this PR fixes** 🔨
AP-515

**Special notes for reviewers** 🗒️
Test with `cmd+z` (ctrl on win) and `cmd+shift+z` and also with mouse right click and then undo/redo

**Additional comments** 🎤

### CODE REVIEW

File: components/mixins/UI/WithHistory.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
+// @ts-nocheck
+import { Dayjs } from 'dayjs'
+import Cursor from '~/libraries/ui/Cursor'
+import { Config } from '~/config'
```
Comment (stavares843) [inline components/mixins/UI/WithHistory.ts:4]: for the imports to be organized they should be displayed as 

```
import { Dayjs } from 'dayjs'
import { Config } from '~/config'
import Cursor from '~/libraries/ui/Cursor'
```

**Comment by josephmcg** [comment]
Is there a reason you didn't extend Vue on the mixin? Just curious because I recently extended Vue on the ContextMenu to clear up error messages, maybe I'll need to go back lol

**Comment by molimauro** [comment]
> Is there a reason you didn't extend Vue on the mixin? Just curious because I recently extended Vue on the ContextMenu to clear up error messages, maybe I'll need to go back lol
> 
> I really like the feel of it, the time interval is brilliant

Actually no and it's even better so i updated it as well. Thanks! :)

**Review by WanderingHogan — APPROVED** [review]

**Review by stavares843 — APPROVED** [review]

**Pull request merged**